File: Models/Entities/Guest.cs
namespace HMS.Models.Entities;

public class Guest
{
    public Guest()
    {
        Bookings = new List<Booking>();
    }

    public int Id { get; set; }

    public string? FullName { get; set; }

    public string? PhoneNumber { get; set; }

    public string? Email { get; set; }

    public DateTime CheckInDate { get; set; }

    public DateTime? CheckOutDate { get; set; }

    public List<Booking> Bookings { get; set; }
}

File: client-app/src/pages/Guest.js
import React , {useState, useEffect, Fragment} from 'react'
import Table from 'react-bootstrap/Table';

const GUEST = () => {

    //Here we will store for api response for all Guest 
    const [data, setData] = useState([]);

    useEffect(() => {
        fetch(window.ApiBasePath + "/Guest")
            .then((res) => res.json())
            .then((result) => {
                const list = result || [];
                setData(list);
            });
    }, [])

    return(
        <Fragment>
            <Table>
      <thead>
        <tr>
          <th>#</th>
          <th>Full Name</th>
          <th>Phone Number</th>
          <th>Email</th>
          <th>Check In Date</th>
          <th>Check Out Date</th>
        </tr>
      </thead>
      <tbody>
        {
            data && data.length > 0 ?
                data.map((item, index) => {
                    return (
                        <tr key={index}>
                        <td>{index + 1}</td>    
                        <td>{item.fullName}</td>
                        <td>{item.phoneNumber}</td>
                        <td>{item.email}</td>
                        <td>{item.checkInDate}</td>
                        <td>{item.checkOutDate}</td>
                      </tr>
                    );
                })
                :
                'Loading...'
        }
       
        
      </tbody>
    </Table>
        </Fragment>
    )
}

export default GUEST;